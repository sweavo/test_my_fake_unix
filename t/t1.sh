#!/bin/bash
## with set -e in force, I expect a loop to abandon the whole script if 
## a command inside the loop exits with non-zero.  In MSys, the loop
## exits, but the script continues to success.

function exit_handler 
{
    echo exit_status $?
    echo done
}
trap exit_handler EXIT

function nonzero_return_function
{
    return 1
}

set -e # exit on error
echo loop_entry

echo hello | while read a 
do
    nonzero_return_function || exit 1
    echo loop_continued
done

echo loop_exit $?

