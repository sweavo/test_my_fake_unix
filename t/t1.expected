loop_entry
exit_status 1
done
